BUG: When changing SQLFilter parameter, resulting SQL query is not generated correctly

dbannik opened the pull request:
### Is not correctly generated sql when changed/switched sqlFilter parameters

`CachedPersisterContext::$selectJoinSql` should be clear or regenerated when sqlFilter changed

The problem reproduce when in use `fetch=EAGER` and use additional sql filter on this property

greg0ire commented on tests/Tests/ORM/Functional/Ticket/SwitchContextWithFilter/User.php:
```diff
@@ -0,0 +1,34 @@
+/**
+ * @ORM\Entity
+ */
+class User
```
`User` is a reserved keyword in PostgreSQL. You should probably use a `Table` annotation here as well because of that.

dbannik committed:
Is not correctly generated sql when changed/switched sqlFilter parameters

CachedPersisterContext::$selectJoinSql should be clear or regenerated when sqlFilter changed
The problem reproduce when in use fetch=EAGER and use additional sql filter on this property

greg0ire commented:
:thinking: why is this targeting 2.21.x?

dbannik commented:
@greg0ire I don't know which branch to merge into 
in my case, in version 2.19 and higher, this problem is present

greg0ire commented:
Take a look at https://github.com/doctrine/orm/issues/11208

dbannik commented:
@greg0ire I sent it to `2.20` and prepared fixes https://github.com/doctrine/orm/pull/11695 for `3.3` and higher

greg0ire commented:
Thanks @dbannik !
